engine/src/main/java/com/redhat/cloud/notifications/events/EndpointProcessor.java

```diff
@@ -93,7 +80,11 @@ private EndpointTypeProcessor endpointTypeToProcessor(EndpointType endpointType)
             case EMAIL_SUBSCRIPTION:
                 return emails;
             default:
-                return (action, endpoints) -> Collections.emptyList();
+                return new EndpointTypeProcessor() {
+                    @Override
+                    public void process(Event event, List<Endpoint> endpoints) {
+                    }
```

Comment:
Wondering if we should log this or remove the `default` altogether (IIUC the code would fail to compile if we add a new `EndpointType`) and add the DEFAULT endpoint type and log if it is still around.

Reply:
I think we could throw an exception here, something like
```
default:
    throw new IllegalArgumentException("Unexpected endpoint type: " + endpointType);
```

Reply:
I added the exception throw. Now that each endpoint type is isolated within a try-catch block, it's okay to throw an exception as it won't have any impact on other endpoint types.